applications/StructuralMechanicsApplication/python_scripts/symbolic_generation/total_lagrangian_mixed_volumetric_strain_element/total_lagrangian_mixed_volumetric_strain_element_template.cpp

```diff
@@ -0,0 +1,1261 @@
+/***********************************************************************************/
+
+template<>
+void TotalLagrangianMixedVolumetricStrainElement<2>::CalculateEquivalentF(KinematicVariables& rThisKinematicVariables) const
+{
+    // Define references to the auxiliary symbols
+    const auto& N = rThisKinematicVariables.N;
+    const auto& DN = rThisKinematicVariables.DN_DX;
+    const auto& u = rThisKinematicVariables.Displacements;
+    const auto& th = rThisKinematicVariables.JacobianDeterminant;
+    auto& r_eq_def_gradient = rThisKinematicVariables.F;
+    double& r_det_eq_def_gradient = rThisKinematicVariables.detF;
+
+    // Fill the equivalent deformation gradient values
+    //substitute_def_gradient_2D_3N
+    //substitute_det_def_gradient_2D_3N
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<>
+void TotalLagrangianMixedVolumetricStrainElement<3>::CalculateEquivalentF(KinematicVariables& rThisKinematicVariables) const
+{
+    // Define references to the auxiliary symbols
+    const auto& N = rThisKinematicVariables.N;
+    const auto& DN = rThisKinematicVariables.DN_DX;
+    const auto& u = rThisKinematicVariables.Displacements;
+    const auto& th = rThisKinematicVariables.JacobianDeterminant;
+    auto& r_eq_def_gradient = rThisKinematicVariables.F;
+    double& r_det_eq_def_gradient = rThisKinematicVariables.detF;
+
+    // Fill the equivalent deformation gradient values
+    //substitute_def_gradient_3D_4N
+    //substitute_det_def_gradient_3D_4N
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<std::size_t TDim>
+int  TotalLagrangianMixedVolumetricStrainElement<TDim>::Check(const ProcessInfo& rCurrentProcessInfo) const
+{
+    KRATOS_TRY
+
+    int check = TotalLagrangianMixedVolumetricStrainElement::BaseType::Check(rCurrentProcessInfo);
+
+    // Base check
+    check = StructuralMechanicsElementUtilities::SolidElementCheck(*this, rCurrentProcessInfo, mConstitutiveLawVector);
+
+    // Check that the element's nodes contain all required SolutionStepData and Degrees of freedom
+    const auto& r_geometry = this->GetGeometry();
+    for ( IndexType i = 0; i < r_geometry.size(); i++ ) {
+        const NodeType& r_node = r_geometry[i];
+        KRATOS_CHECK_VARIABLE_IN_NODAL_DATA(VOLUMETRIC_STRAIN,r_node)
+        KRATOS_CHECK_DOF_IN_NODE(VOLUMETRIC_STRAIN, r_node)
+    }
+
+    return check;
+
+    KRATOS_CATCH( "" );
+}
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+template<std::size_t TDim>
+void TotalLagrangianMixedVolumetricStrainElement<TDim>::CalculateOnIntegrationPoints(
```

Comment:
There are many variables not computed, I would add more as in the base solid (which is not the base class of this)

Reply:
It is intentionally done like this. Indeed this was already discussed when implementing the small displacements one. It makes no sense to derive from the base solid element as the kinematics are completely different in this case, note that we are using an enriched strain definition, and it would result in an unnecessarily complexity addition.

Reply:
I am not saying to derive from it, but adding more terms as the ones there

Reply:
(variables)

Reply:
what do you see as missing? in any case i would first merge this  and later add this, once we already have test coverage

Reply:
Ah, I missed @loumalouomega 's point. What you ask for is to give support to more variables to be output. 

After having gone through current variables and the ones in the base element, I tend to agree @RiccardoRossi on this, I think we are already outputting the most used ones. In any case, and as a compromise solution, for the moment I'll add the ones already present in the GUI that are not here, that are the Von Mises stress and the Almansi strain vector. The rest I'd add them upon request.